kng.go

```diff
@@ -309,8 +312,82 @@ func (k *Kafka) getSaramaConfig() *sarama.Config {
 	return cfg
 }
 
+func (k *Kafka) GetNextOffset(ctx context.Context, topic, consumerGroup string) (int64, error) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "kafka.GetNextOffset")
+	defer span.Finish()
+
+	c, err := sarama.NewClient(k.Options.Brokers, k.getSaramaConfig())
+	if err != nil {
+		return 0, errors.Wrap(err, "unable to create kafka client")
+	}
+
+	om, err := sarama.NewOffsetManagerFromClient(consumerGroup, c)
+	if err != nil {
+		return 0, errors.Wrap(err, "unable to create offset manager client")
+	}
+
+	partitions, err := c.Partitions(topic)
+	if err != nil {
+		return 0, errors.Wrap(err, "unable to list partitions")
+	}
+
+	var nextOffset int64 = 0
+
+	for _, partition := range partitions {
+		p, err := om.ManagePartition(topic, partition)
+		if err != nil {
+			return 0, errors.Wrap(err, "unable to get partition manager")
+		}
+
+		res, _ := p.NextOffset()
+
+		if res > nextOffset {
+			nextOffset = res
+		}
+	}
+
+	return nextOffset, nil
+}
+
+func (k *Kafka) CreateConsumerGroup(ctx context.Context, topic, consumerName string, startOffset int64) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "kafka.CreateConsumerGroup")
+	defer span.Finish()
+
+	// TODO: figure these options out
+	cfg := kafka.ConsumerGroupConfig{
+		ID:                    consumerName,
+		Brokers:               k.Options.Brokers,
+		Dialer:                k.Dialer,
+		Topics:                []string{topic},
+		WatchPartitionChanges: true,
+		GroupBalancers:        nil,
+		RetentionTime:         -1,
+		StartOffset:           startOffset,
```

Comment:
According to the docs, the only "valid" thing here is `FirstOffset` or `LastOffset`. I will look at code to see if this actually uses any other provided value.

Reply:
Update: don't think this will work because of:

https://github.com/segmentio/kafka-go/blob/eba9cae7fd57401a8078f6f25bc00cb753cd4f42/consumergroup.go#L247-L249

Will research on how to achieve this.